generator/src/main/java/com/scottlogic/deg/generator/validators/ContradictionChecker.java

```diff
@@ -0,0 +1,72 @@
+package com.scottlogic.deg.generator.validators;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.common.profile.constraints.atomic.AtomicConstraint;
+import com.scottlogic.deg.generator.decisiontree.ConstraintNode;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.reducer.ConstraintReducer;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ContradictionChecker {
+    private final ConstraintReducer constraintReducer;
+
+    @Inject
+    public ContradictionChecker(ConstraintReducer constraintReducer){
+        this.constraintReducer = constraintReducer;
+    }
+    /**
+     * Check for any contradictions
```

Comment:
this javadoc seems fairly pointless in its current state.

expand or delete it